airflow/www/templates/airflow/dags.html

```diff
@@ -98,7 +98,10 @@ <h2>{{ page_title }}</h2>
               </th>
               <th>Schedule</th>
               <th style="width:180px;">Last Run
-                <span class="material-icons text-muted js-tooltip" aria-hidden="true" title="Execution Date/Time of Highest Dag Run.">info</span>
+                <span class="material-icons text-muted js-tooltip" aria-hidden="true" title="Logical Date/Time of the latest Dag Run.">info</span>
+              </th>
+              <th style="width:180px;">Next Run
+                <span class="material-icons text-muted js-tooltip" aria-hidden="true" title="Expected Date/Time of the next Dag Run.">info</span>
```

Comment:
What date does this show? Is it the logical date that is going to be created, or the time _when_ the run will be created?

I think we need to display both of these (though one should be in a tooltip I think)

Reply:
Good point, it’s the logical date I think. The next expected *run* time is available in `DagModel.next_dagrun_create_after`.

This raises another issue though—perhaps we should ditch all the logical date labels, and always display the current/next DAG run’s run date and data interval instead.

Reply:
@uranusjr Hmmm that might remove the ambiguity, yeah

Do we retroactively set data_interval on old runs when in our migration?

Reply:
We don’t, but those can be calculated on the fly rather cheaply when needed. Or we could just display `data_interval_start` as *Unknown* (perhaps with a tooltip explaining why).

Reply:
Or something like

```jinja
{% if dagrun.data_interval_start is not None or dagrun.data_interval_end is not None %}
    {# Render data interval #}
{% else %}
    {# Render logical date #}
{% endif %}
```

Since users probably would want to see `execution_date` for those old runs?

Reply:
Right now we get the last run execution date and start time here: https://github.com/apache/airflow/blob/main/airflow/www/views.py#L919

How should we render it with data_interval?

Reply:
Oh… that’s a bit complicated due to how `query` is constructed… I’ll add a commit to your branch for this later (need to do something first).

Reply:
Added, I think this should work.

Reply:
Pushed, I think this should work…

Reply:
Actually, we use `execution_date` in a lot more parts of the UI. It should be a separate PR.